modules/mi_datagram/mi_datagram_writer.c

```diff
@@ -297,8 +295,6 @@ int mi_datagram_flush_tree(datagram_stream * dtgram, struct mi_root *tree)
 		return -1;
 	}
 
-	*(dtgram->current) = '\n';
-	dtgram->len--;
```

Comment:
The idea with this extra `\n` is to have it as End OF Message - I know it makes no sense as it is datagram, but the module shares the same grammar as mi_fifo (which is stream oriented). So you need to have the empty line at the end of the whole response text. So, the final writing should keep this `\n`, but not during the flushing.....I guess only the lines 208-209 should be removed and these ones kept.

Reply:
> So you need to have the empty line at the end of the whole response text.
> So, the final writing should keep this `\n`, but not during the flushing.....
> I guess only the lines 208-209 should be removed and these ones kept.

Okay, I can agree to that, but then we need to `(dtgram->current++)`, otherwise we'd replace the LF with NUL on (original) line 302.

Reply:
Force pushed and updated.

Reply:
:+1: